### PR TITLE
Exlucding inpudata.shadowMask if ShadowMask is not used

### PR DESCRIPTION
**Pull request by lukaschod**
# **Please read;**
## **PR Workflow for the Graphics repository:**
* **All PRs must be opened as draft initially**
* Reviewers can be added while the PR is still in draft
* The PR can be marked as “Ready for Review” once the reviewers have confirmed that **no more changes are needed**
* Tests will start automatically after the PR is marked as “Ready for Review”
* **Do not use [skip ci]** - this can break some of our tooling
* Read the [Graphics repository & Yamato FAQ](http://go/graphics-yamato-faq).

### Checklist for PR maker
- [ ] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [ ] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR. If you do add documentation, make sure to add the relevant Graphics Docs team member as a reviewer of the PR. If you are not sure which person to add, see the [Docs team contacts sheet](https://docs.google.com/spreadsheets/d/1rgUWWgwLFEHIQ3Rz-LnK6PAKmbM49DZZ9al4hvnztOo/edit#gid=1058860420).
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
Fixing regression when ShadowMask enabled will break custom shaders that do not have inputdata.shadowMask initialized. This is more like a workaround solution where we avoid susing this new input when shadowmask keyword is not used. This way shader strips it.
For term solution will require more investgation, how we can extend InputData without regressions https://jira.unity3d.com/browse/URP-242.

---
### Testing status
Not something that can be tested trivialy. Elvar working on shader API test validation that would cover this case in the future.

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.universal/ShaderLibrary/Lighting.hlsl [hunk 1]
```diff
@@ -792,6 +792,18 @@ half3 VertexLighting(float3 positionWS, half3 normalWS)
     return vertexLightColor;
 }
 
+half4 GetShadowMask(InputData inputData)
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/ShaderLibrary/Lighting.hlsl:795]: Remove this function. It's not used anywhere anymore. 
Suggestion for follow up PR: provide a GetShadowMask(float2 lightmapUV)

File: TestProjects/UniversalGraphicsTest/Assets/Scenes/023_Lighting_Mixed_Indirect.unity [hunk 2]
```diff
@@ -252,7 +252,7 @@ MonoBehaviour:
     PerPixelCorrectnessThreshold: 0.005
     PerPixelGammaThreshold: 0.003921569
     PerPixelAlphaThreshold: 0.003921569
-    AverageCorrectnessThreshold: 0.001
+    AverageCorrectnessThreshold: 0.0012
```
Comment (phi-lira) [inline TestProjects/UniversalGraphicsTest/Assets/Scenes/023_Lighting_Mixed_Indirect.unity:255]: curious of what we needed to bump the threshold with this PR.

**Review by phi-lira — CHANGES_REQUESTED** [review]

File: TestProjects/UniversalGraphicsTest/Assets/Scenes/023_Lighting_Mixed_Indirect.unity (hunk 2 above)
Comment (lukaschod) [reply TestProjects/UniversalGraphicsTest/Assets/Scenes/023_Lighting_Mixed_Indirect.unity:255]: With Nicolaj we noticed that there is some noise in these baked lighting scenes (Maybe that is why they were disabled in first place, as I enabled them again)

File: com.unity.render-pipelines.universal/ShaderLibrary/Lighting.hlsl (hunk 1 above)
Comment (lukaschod) [reply com.unity.render-pipelines.universal/ShaderLibrary/Lighting.hlsl]: Fixed

**Review by phi-lira — APPROVED** [review]

**Review by ellioman — APPROVED** [review]
Looks good.

**Comment by ellioman** [comment]
The Yamato Failure on DX11 Standalone, is a known timeout issue also present on master and not caused by this PR.

**Pull request merged**